pkg/syncer/types/commontypes.go

```diff
@@ -33,12 +33,12 @@ type ConfigInfo struct {
 }
 
 var (
-	// VirtualCenter object for syncer
-	vcenter *cnsvsphere.VirtualCenter
+	// VirtualCenter instance for syncer
+	vCenterInstance *cnsvsphere.VirtualCenter
 	// Ensure vcenter is a singleton
-	onceForVirtualCenter sync.Once
-	// vcerror message from GetVirtualCenterInstance
-	vcerror error
+	vCenterInitialized bool
```

Comment:
On second thought, one of the reasonings behind removing the `once.Do` pattern was because this function is called within `ReloadConfiguration()` and we want `vCenterInstance` to be reassigned. Adding this variable doesn't solve that problem

Reply:
Good point. But this function is also being called from a few init functions and custom controllers, and we wouldn't want to re-instantiate `vCenterInstance` on every call.
We can add another bool argument to the function indicating whether a reload is required or not if it's already initialized.

Reply:
Handled this is new commit.